lodge.py

```diff
@@ -0,0 +1,127 @@
+import os
+import json
+import sys
+import logging
+
+
+__version__ = "0.1.0"
+
+# To be used by tests
+DEFAULT_LOG_STREAM = sys.stderr
+
+
+def _get_log_level_from_env_var(name: str) -> str:
+    DEFAULT_LOG_LEVEL = os.environ.get("LOG_LEVEL", "INFO")
+    uppercase_underline = "_".join(name.split(".")).upper()
+    env_var_key = f"{uppercase_underline}_LOG_LEVEL"
+    return os.environ.get(env_var_key, DEFAULT_LOG_LEVEL)
+
+
+def _get_default_format():
+    DEFAULT_LOG_ENV = os.environ.get("LOG_ENV", "PROD")
+    LOG_BASE_FIELDS = eval(
+        os.environ.get(
+            "LOG_BASE_FIELDS",
+            '{"message":"%(message)s","timestamp":"%(asctime)s","level":"%(levelname)s",}'
+        )
+    )
+    LOG_EXTRA_FIELDS = eval(os.environ.get("LOG_EXTRA_FIELDS", '{}'))
+
+    if DEFAULT_LOG_ENV == "PROD":
+        log_format = json.dumps({**LOG_BASE_FIELDS, **LOG_EXTRA_FIELDS})
+    else:
+        log_format = "%(asctime)s | %(levelname)s | %(name)s | %(message)s"
+
+    return logging.Formatter(log_format)
+
+
+def _add_base_configs(logger: logging.Logger):
+    logging.addLevelName(logging.CRITICAL, "FATAL")
+    logging.addLevelName(logging.WARNING, "WARN")
+
+
+def _add_default_handler(logger: logging.Logger):
+    stderr = logging.StreamHandler(DEFAULT_LOG_STREAM)
+    stderr.setFormatter(_get_default_format())
+    logger.addHandler(stderr)
+
+
+# https://github.com/python/cpython/blob/e8e341993e3f80a3c456fb8e0219530c93c13151/Lib/logging/__init__.py#L162
+if hasattr(sys, '_getframe'):
+    currentframe = lambda level: sys._getframe(level)  # noqa
+else:  # pragma: no cover
+    def currentframe(level: int):
+        """Return the frame object for the caller's stack frame."""
+        try:
+            raise Exception
+        except Exception:
+            return sys.exc_info()[2].tb_frame.f_back  # type: ignore
+
+
+def get_logger(name: str) -> logging.Logger:
```

Comment:
He is not encouraged, but it can use it. In the case where he wants to create a log that doesn't depend on the package and modules structure.

I thought on exposing handlers and formatting, to be able to change when calling this functions. But didn't do this to not clutter the implemention. We can add it later